cluster-provision/images/vm-image-builder/build-containerdisk.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+SCRIPT_DIR="$(
+  cd "$(dirname "$BASH_SOURCE[0]")"
+  pwd
+)"
+
+function cleanup() {
+  echo "[build-container-disk] cleanup"
+  rm -f Dockerfile
+
+  if [ $? -ne 0 ]; then
+    rm -f "$tar_file"
+    docker image rm "$full_image_tag"
+  fi
```

Comment:
I do not understand this one. If the file removal done earlier fails, you do not remove the rest?
It also uses "-e" so it is expected to fail anyway on first error, I see no point to check for the returned value.

Reply:
The idea is to  delete the exported image tar file and the image stored in local docker registry, if there is was error during  he script.
I will move lines 13-16 to line 10 so it will actually catch errors from the script and not from cleanup().

Reply:
I'm not sure I understand what you just mentioned but please make sure to test your conditions to make sure they do what you indented them to do.